script/index.coffee

```diff
@@ -1463,9 +1466,16 @@ api.wrap = (user, main=true) ->
               thatDay = moment(now).subtract({days: n + 1})
               scheduleMisses++ if api.shouldDo(thatDay, repeat, user.preferences)
           if scheduleMisses > 0
-            perfect = false if type is 'daily'
+            if type is 'daily'
+              perfect = false 
+              if task.checklist?.length > 0  # Partially completed checklists dock less mana points
+                dailyDueUnchecked += (1 - _.reduce(task.checklist,((m,i)->m+(if i.completed then 1 else 0)),0) / task.checklist.length)
+              else
+               dailyDueUnchecked += 1
             delta = user.ops.score({params:{id:task.id, direction:'down'}, query:{times:scheduleMisses, cron:true}});
             user.party.quest.progress.down += delta if type is 'daily'
+        else
+          dailyChecked++
```

Comment:
You're right. Thanks for catching this Alys. I'll fix it and change to an
if-else statement later today.

On Sat, Jan 24, 2015 at 3:13 AM, Alice Harris <notifications@github.com>
wrote:

> In script/index.coffee
> <https://github.com/HabitRPG/habitrpg-shared/pull/448#discussion_r23495017>
> :
>
> >              delta = user.ops.score({params:{id:task.id, direction:'down'}, query:{times:scheduleMisses, cron:true}});
> >              user.party.quest.progress.down += delta if type is 'daily'
> > +        else
> > +          dailyChecked++
>
> @Verabird <https://github.com/Verabird> Is dailyChecked incremented by
> completed To-Dos? I *think* it is from reading the code, but I can't test
> because I've broken my local install (fourth time in three days).
>
> If you need to change the code for this, I think it's worth also reversing
> the unless completed .... else .... statement to make it if-else. Now
> that we have an else branch, there's no pressing need to use unless, and
> the logic will be easier to follow if it's an if-else statement (especially
> since we have several programmers who aren't very familiar with
> coffeescript).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HabitRPG/habitrpg-shared/pull/448/files#r23495017>.
>